controllers/reaper_controller_test.go

```diff
@@ -149,35 +149,60 @@ var _ = Describe("Reaper controller", func() {
 
 		By("check that the deployment is created")
 		deploymentKey := types.NamespacedName{Namespace: ReaperNamespace, Name: ReaperName}
-		deployment := &appsv1.Deployment{}
+		currentDeployment := &appsv1.Deployment{}
 
 		Eventually(func() error {
-			return k8sClient.Get(context.Background(), deploymentKey, deployment)
+			return k8sClient.Get(context.Background(), deploymentKey, currentDeployment)
 		}, timeout, interval).Should(Succeed(), "deployment creation check failed")
 
-		Expect(len(deployment.OwnerReferences)).Should(Equal(1), "deployment owner reference not set")
-		Expect(deployment.OwnerReferences[0].UID).Should(Equal(reaper.GetUID()), "deployment owner reference has wrong uid")
+		Expect(len(currentDeployment.OwnerReferences)).Should(Equal(1), "deployment owner reference not set")
+		Expect(currentDeployment.OwnerReferences[0].UID).Should(Equal(reaper.GetUID()), "deployment owner reference has wrong uid")
 
 		By("update deployment to be ready")
-		patchDeploymentStatus(deployment, 1, 1)
+		patchDeploymentStatus(currentDeployment, 1, 1)
 
 		verifyReaperReady(types.NamespacedName{Namespace: ReaperNamespace, Name: ReaperName})
 
+		// Simulate changing the deployment strategy to rolling.
+		// The deployment strategy should be reconciled back to recreate.
+		By("patch deployment to have a rolling update strategy")
```

Comment:
A small question. The new deployment sets the strategy, but existing ones have nothing. Will the reconciler still catch the existing deployments without preset one?

Reply:
Reaper's deployments are created with Recreate by default as per [here](https://github.com/Miles-Garnsey/reaper-operator/blob/miles/deployment-strategy-recreate/pkg/reconcile/reconcilers.go#L519).

Reply:
That's only with new deployments. That part of the code wasn't there in the old reaper-operator.

Reply:
The default strategy is rolling. If you check the logic it checks whether the deployment strategy is [not recreate](https://github.com/Miles-Garnsey/reaper-operator/blob/miles/deployment-strategy-recreate/pkg/reconcile/reconcilers.go#L289).

I don't feel like I'm understanding your question here, please clarify with references to the code.